partition_snapshot_reader.hh

```diff
@@ -133,29 +136,46 @@ class partition_snapshot_flat_reader : public flat_mutation_reader::impl, public
             return e;
         }
 
-        const range_tombstone& pop_range_tombstone() {
+        range_tombstone pop_range_tombstone() {
             boost::range::pop_heap(_range_tombstones, _heap_cmp);
             auto& current = _range_tombstones.back();
-            const range_tombstone& rt = *current.begin();
-            current.advance_begin(1);
+            range_tombstone rt = *(Reversing ? std::prev(current.end()) : current.begin());
+            if constexpr (Reversing) {
+                current.advance_end(-1);
+            } else {
+                current.advance_begin(1);
+            }
             if (current.begin() == current.end()) {
                 _range_tombstones.pop_back();
             } else {
                 boost::range::push_heap(_range_tombstones, _heap_cmp);
             }
+            if constexpr (Reversing) {
+                rt.reverse();
+            }
             return rt;
         }
 
         // Valid if has_more_rows()
         const rows_entry& peek_row() const {
-            return *_clustering_rows.front()._position;
+            const auto& r = _clustering_rows.front();
+            if constexpr (Reversing)
+                return *std::prev(r._end);
+            else
+                return *r._position;
         }
```

Comment:
Consider using the reverse iterator of the containers. Maybe it ends up simpler.

Reply:
Perhaps we can have a reversed_range_tombstone_list that wraps a range_tombstone_list and returns the reversed iterators and does the reversal by itself. Probably not worth the effort given we're changing the representation with mutation_fragment_v2.